primitives/runtime/src/generic/checked_extrinsic.rs

```diff
@@ -36,28 +36,26 @@ pub struct CheckedExtrinsic<AccountId, Call, Extra> {
 	pub function: Call,
 }
 
-impl<AccountId, Call, Extra, Origin, Info> traits::Applyable for
+impl<AccountId, Call, Extra, Origin> traits::Applyable for
 	CheckedExtrinsic<AccountId, Call, Extra>
 where
 	AccountId: Member + MaybeDisplay,
 	Call: Member + Dispatchable<Origin=Origin>,
-	Extra: SignedExtension<AccountId=AccountId, Call=Call, DispatchInfo=Info>,
+	Extra: SignedExtension<AccountId=AccountId, Call=Call>,
 	Origin: From<Option<AccountId>>,
-	Info: Clone,
 {
 	type Call = Call;
-	type DispatchInfo = Info;
 
 	fn validate<U: ValidateUnsigned<Call = Self::Call>>(
 		&self,
 		// TODO [#5006;ToDr] should source be passed to `SignedExtension`s?
 		// Perhaps a change for 2.0 to avoid breaking too much APIs?
 		source: TransactionSource,
-		info: Self::DispatchInfo,
+		info: &<Self::Call as Dispatchable>::Info,
```

Comment:
We might create `type DispatchInfoFor<T> = <<T as Applyable>::Call as Dispatchable>::Info` to avoid these duplications.

Reply:
`DispatchInfoOf` to be more substrate-idiomatic? :D

Reply:
But where to put these aliases. Every implementer of `SignedExtension` need those. Just put these in `sp_runtime::traits`?:
```
pub type CallOf<T> = <T as SignedExtension>::Call;
pub type DispatchInfoOf<T> = <CallOf<T> as Dispatchable>::Info;
pub type PostDispatchInfoOf<T> = <CallOf<T> as Dispatchable>::PostInfo
```
But then the user must import them. This is annoying, too?

Reply:
I don't think it can get any better and fine by me. Just document them as: _use this if you want to_.

Reply:
Imho less annoying to import than to bake in the bounds again. Also the `<.. as ...>` syntax looks uglier to me. But not super strong on this, so whatever you prefer.

Reply:
I added the type aliases and used them in the existing code to get rid of the ugly casting.